src/Metadata/Property/SubresourceMetadata.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Metadata\Property;
+
+/**
+ * Subresource metadata.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class SubresourceMetadata
```

Comment:
It does and in the end we'll have less function calls to retrieve the same information.

```php
           $isCollection = $propertyMetadata->getType()->isCollection();
           $subresource = $isCollection ? $propertyMetadata->getType()->getCollectionValueType()->getClassName() : $propertyMetadata->getType()->getClassName();
```

This will also ease the maintenance to have a SubresourceMetadata class if we need to support subresource related cases in the future no?